### PR TITLE
Blocks

### PR DESCRIPTION
**Pull request by Pega88**
Blocks section.

### CODE REVIEW

File: en/4-blocks/1-Introduction_to_blocks_and_blockchain.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Introduction to blocks & blockchain
+
+## Concept
+When Alice wants to send a transaction to Bob, she will sign her transaction and broadcast it to the nodes in the network. The consensus nodes will select transactions that they will finalise, and include them into a block. A block is a batch of transactions that get confirmed and stored at the same time. Apart from all transactions that a certain block entails, there are some other fields present in each block which are discussed [here](2-Structure_of_a_block.md).
+
+## Chaining the blocks
+One of these fields in each block (b<sub>n</sub>) is a reference to the previous block that was just confirmed before this block (b<sub>n-1</sub>). In more detail, is is the hash of the header of the previous block. It is up to the nodes to calculate the hash to validate the parent. In its turn, this previous block(b<sub>n-1</sub>) has the same behaviour, linking to the block before it(b<sub>n-2</sub>). This chain goes all the way back to the first block ever created, called the Genesis (b<sub>0</sub>) block. This is where the term 'blockchain' comes from, as it is, in fact, a chain of blocks.
```
Comment (Qiao-Jin) [inline en/4-blocks/1-Introduction_to_blocks_and_blockchain.md]: is is the hash of the header of the previous block ->
it is the hash of the header of the previous block

File: en/4-blocks/2-Structure_of_a_block.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+## Block body
+
+
+The body of a block contains all the transactions that are embodied in that block.
+
+### Transaction types
+Below are the type of transactions that can occur in a block:
+* **MinerTransaction**: The first transaction in each block must be MinerTransaction. It is used to reward all transaction fees of the current block to the validator.
+* **IssueTransaction**: Used to issue an asset. Asset managers can create the assets that have been registered in NEO's block chain through IssueTransaction, and sent them to any address.
```
Comment (Qiao-Jin) [inline en/4-blocks/2-Structure_of_a_block.md]: sent them to any address ->
send them to any address

File: en/4-blocks/2-Structure_of_a_block.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+* **ClaimTransaction**: Transaction invoked to claim NeoGas thas was not yet claimed for holding NEO.
```
Comment (Qiao-Jin) [inline en/4-blocks/2-Structure_of_a_block.md]: thas -> that

File: en/4-blocks/3-Block_creation_broadcasting.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
+# Block creation
+## Consensus nodes
+As discussed in [Part 1](1-Introduction_to_blocks_and_blockchain.md), Neo uses the dBFT mechanism to generate new blocks. In short, this means there are a select amount of consensus nodes, elected by the network.
```
Comment (Qiao-Jin) [inline en/4-blocks/3-Block_creation_broadcasting.md]: select -> selected

File: en/4-blocks/3-Block_creation_broadcasting.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+This follows from the fact that a UTXO is not divisible and needs to be spent in a whole. For example, when an UTXO has a value of 500 GAS, spending that UTXO will spend the full 500 GAS. When the transaction is meant to send only 5 GAS, the other 495 GAS need to be send back to yourself, accounting for the 'change'. Compare this to handing the cashier a $100 bill when you only need to pay 1$. You expect 99$ change in return. With an UTXO, this is similar. You need to pay this change back to yourself in the same transaction.
```
Comment (Qiao-Jin) [inline en/4-blocks/3-Block_creation_broadcasting.md]: For example, when an UTXO has a value of 500 GAS, spending that UTXO will spend the full 500 GAS. When the transaction is meant to send only 5 GAS, the other 495 GAS need to be send back to yourself, accounting for the 'change' ->
For example, you want to send a friend 5 GAS but only have one single 500-GAS UTXO in pocket. You cannot split that 500-GAS UTXO directly. What you will do is to create a transaction with that mentioned UTXO as input. You will add to the transaction 2 outputs as well: 5 GAS to your friend while the remaining 495 back to yourself.

File: en/4-blocks/3-Block_creation_broadcasting.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+As an example, you can look at [this specific transaction](https://neoscan.io/transaction/80b963d68c0f3d65c0e906057f7517a17ef7dcc1b29e2a79205e4aa235131f95).
+
+![Transaction example with fee](txn-with-fee.png)
+
+The source address had an UTXO containing 119.99999987 GAS and wanted to send 60 GAS to another address. To do so, it specified the recipient together with the 60 GAS, and it spent the remaining GAS to itself again (59.99999986 GAS). However, 60 + 59.99999986 equals to 119.99999986 GAS. This differs with 0.00000001 GAS, which was the implied transaction fee. By not specifying the address for the 0.00000001 that was left from the UTXO, it was handled as transaction fee for the consensus node that created [the related block](https://neoscan.io/block/eabb36acfb055592afb06786f831390762000a9aa669016d27eb031f41e14a07).
+
+
+## Other fees
+A transaction fee is used to gain priority on the network. There are however fees with specific puroses.
```
Comment (Qiao-Jin) [inline en/4-blocks/3-Block_creation_broadcasting.md]: puroses -> purposes

File: en/4-blocks/3-Block_creation_broadcasting.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+- ***Smart Contract Creation*** The current free to deploy a new contract on the Mainnet is 500 GAS. For structured development, it is advised to start development on a local testnet (link docker). Once the Smart Contract is stable, you can apply for Testnet funds here (link), for final validations. Once you are certain your Smart Contract is implemented correctly, only then should you deploy it onto the Mainnet, as it is irrevokable - meaning you will not get your 500 GAS back even when you destroy the contract.
```
Comment (Qiao-Jin) [inline en/4-blocks/3-Block_creation_broadcasting.md]: free -> fee

File: en/4-blocks/4-Block_validation_processing.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+|0x43df832e612d220c3ce639cbca4ab18f9ce21dfd8718340dae4ca49fd239a4ee06d294e9d583bc4e0da5cccb5eb0f35e7b836d7c633b1e9ca20c0c0af4296447|
+|0xc352f2425c56001a292cc28cba9e736913f9c116580796efacda4270ac104659173ee1bc4bed706c0f3ea90c9b8201653abe74a30e627c443855af6c08c90acb|
+
+## Interpretation
+We've done it! Now we know what's inside both the Invocation and Verification Script, let's look at what it means. The goal of these scripts is allow any node to validate that the block was actually created and agreed upon by the consensus nodes. To do this, the consensus nodes sign the block with their private key. From the moment we have 5 of the 7 consensus nodes that signed the same block, we achieve consensus and the block is valid. These 5 signatures are the ones included in the Invocation Script.
+
+Now all we need to be able to do, is validate that these 5 signatures are actually valid to validate the block. For that, the block contains the Verification Script. This way, the block tells every node how to validate it. The node that validates the block thus executes the Verification Script, and if this script passes, then the block is marked as valid.
```
Comment (Qiao-Jin) [inline en/4-blocks/4-Block_validation_processing.md]: is validate -> is to validate

Comment (Pega88) [reply en/4-blocks/4-Block_validation_processing.md]: fixed, will push with other fixes.

File: en/4-blocks/1-Introduction_to_blocks_and_blockchain.md (hunk 1 above)
Comment (Pega88) [reply en/4-blocks/1-Introduction_to_blocks_and_blockchain.md]: fixed, will push with other fixes.

File: en/4-blocks/2-Structure_of_a_block.md (hunk 2 above)
Comment (Pega88) [reply en/4-blocks/2-Structure_of_a_block.md]: fixed, will push with other fixes.

File: en/4-blocks/2-Structure_of_a_block.md (hunk 3 above)
Comment (Pega88) [reply en/4-blocks/2-Structure_of_a_block.md]: fixed, will push with other fixes.

File: en/4-blocks/3-Block_creation_broadcasting.md (hunk 4 above)
Comment (Pega88) [reply en/4-blocks/3-Block_creation_broadcasting.md]: fixed, will push with other fixes.

File: en/4-blocks/3-Block_creation_broadcasting.md (hunk 6 above)
Comment (Pega88) [reply en/4-blocks/3-Block_creation_broadcasting.md]: fixed, will push with other fixes.

File: en/4-blocks/3-Block_creation_broadcasting.md (hunk 7 above)
Comment (Pega88) [reply en/4-blocks/3-Block_creation_broadcasting.md]: fixed, will push with other fixes.

File: en/4-blocks/4-Block_validation_processing.md (hunk 8 above)
Comment (Pega88) [reply en/4-blocks/4-Block_validation_processing.md]: fixed, will push with other fixes.

File: en/4-blocks/3-Block_creation_broadcasting.md (hunk 5 above)
Comment (Pega88) [reply en/4-blocks/3-Block_creation_broadcasting.md]: fixed

**Comment by Pega88** [comment]
@Qiao-Jin @steven1227 fixed all comments

**Comment by Qiao-Jin** [comment]
Seems good to me now:)

**Pull request merged**